publisher-sdk/src/main/java/com/criteo/publisher/network/LiveBidRequestSender.kt

```diff
@@ -29,19 +30,17 @@ class LiveBidRequestSender(
     private val pubSdkApi: PubSdkApi,
     private val cdbRequestFactory: CdbRequestFactory,
     private val executor: Executor,
-    private val scheduledExecutorService: ScheduledExecutorService
+    private val scheduledExecutorService: ScheduledExecutorService,
+    private val config: Config
 ) {
-  companion object {
-    const val DEFAULT_TIME_BUDGET_IN_MILLIS = 5_000L
-  }
 
   fun sendLiveBidRequest(
       cacheAdUnit: CacheAdUnit,
       liveCdbCallListener: LiveCdbCallListener
   ) {
     scheduledExecutorService.schedule({
       liveCdbCallListener.onTimeBudgetExceeded()
-    }, DEFAULT_TIME_BUDGET_IN_MILLIS, TimeUnit.MILLISECONDS)
+    }, config.liveBiddingTimeBudgetInMillis.toLong(), TimeUnit.MILLISECONDS)
```

Comment:
can you define the default as a long value to prevent this conversion?

Reply:
I'm afraid I am not understanding you. This is not a default value, this is the value given by CDB.

Using an `int` instead of `long` from CaC to `Config#getLiveBiddingTimeBudgetInMilis` is done in purpose: there is no way that such value could be over 2^32.
IMHO we should not chose a technical type just because of one techno forcing us this cast: ie if `ScheduledExecutorService` accepted an `int` then we wouldn't even think about using a `long`. Moreover going from `int` to `long` is lossless.

Reply:
Ah I read `buildConfig.liveBiddingTimeBudgetInMillis`, nvm.